api/provisioning/v1alpha1/provisioningrequest_validation.go

```diff
@@ -350,3 +350,27 @@ func matchesAnyPattern(path []string, patterns [][]string) bool {
 	}
 	return false
 }
+
+// HasFatalProvisioningFailure checks if the ProvisioningRequest
+// has a fatal provisioning failure that cannot be recovered
+// on its own.
+func HasFatalProvisioningFailure(conditions []metav1.Condition) bool {
+	var conditionTypes = []ConditionType{
```

Comment:
nit: It would be nice to avoid having a hardcoded list of these types, which could end up being stale if a new condition is added but the various places we have such a hardcoded list aren't updated. I think it would be better to either have this list defined in the same place as the PRconditionTypes var-struct, or determine it programmatically. I'm sure there's a nice way to do it, but one option that relies on the struct field name being the same as the string representation would be:
```
for _, f := range reflect.VisibleFields(reflect.TypeOf(PRconditionReasons)) {
```
This `reflect.VisibleFields` returns a list of the field names in the struct, and would avoid hardcoding.

Reply:
Thanks for your suggestion, Don! In our case, this cannot be fully determined programmatically since only a subset of the available condition types are considered fatal. Reflection could enumerate all condition types, but we would still need a way to mark only some as fatal (e.g., struct tag or a metadata map), which is essentially equivalent to defining a list of fatal conditions. 
For simplicity, I will just move this list to the same location where the PRconditionTypes is defined. Hopefully, the condition changes will be infrequent.